Assets/MixedRealityToolkit.SDK/Experimental/Features/Input/Handlers/PrimitiveManipulator.cs

```diff
@@ -0,0 +1,744 @@
+        {
+            get => manipulationType;
+            set => manipulationType = value;
+        }
+
+        [SerializeField]
+        [EnumFlags]
+        [Tooltip("What manipulation will two hands perform?")]
+        private TwoHandedManipulation twoHandedManipulationType = TwoHandedManipulation.Move | TwoHandedManipulation.Rotate | TwoHandedManipulation.Scale;
+        
+        public TwoHandedManipulation TwoHandedManipulationType
+        {
+            get => twoHandedManipulationType;
+            set => twoHandedManipulationType = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Specifies whether manipulation can be done using far interaction with pointers.")]
+        private bool allowFarManipulation = true;
+
+        public bool AllowFarManipulation
+        {
+            get => allowFarManipulation;
+            set => allowFarManipulation = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Rotation behavior of object when using one hand near")]
+        private RotateInOneHandType oneHandRotationModeNear = RotateInOneHandType.RotateAboutGrabPoint;
+
+        public RotateInOneHandType OneHandRotationModeNear
+        {
+            get => oneHandRotationModeNear;
+            set => oneHandRotationModeNear = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Rotation behavior of object when using one hand at distance")]
+        private RotateInOneHandType oneHandRotationModeFar = RotateInOneHandType.RotateAboutGrabPoint;
+
+        public RotateInOneHandType OneHandRotationModeFar
+        {
+            get => oneHandRotationModeFar;
+            set => oneHandRotationModeFar = value;
+        }
+
+        [SerializeField]
+        [EnumFlags]
+        [Tooltip("Rigid body behavior of the dragged object when releasing it.")]
+        private ReleaseBehaviorType releaseBehavior = ReleaseBehaviorType.KeepVelocity | ReleaseBehaviorType.KeepAngularVelocity;
+
+        public ReleaseBehaviorType ReleaseBehavior
+        {
+            get => releaseBehavior;
+            set => releaseBehavior = value;
+        }
+        
+        [SerializeField]
+        [Tooltip("Constrain rotation along an axis")]
+        private RotationConstraintType constraintOnRotation = RotationConstraintType.None;
+
+        public RotationConstraintType ConstraintOnRotation
+        {
+            get => constraintOnRotation;
+            set => constraintOnRotation = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Constrain movement")]
+        private MovementConstraintType constraintOnMovement = MovementConstraintType.None;
+
+        public MovementConstraintType ConstraintOnMovement
+        {
+            get => constraintOnMovement;
+            set => constraintOnMovement = value;
+        }
+        
+        [SerializeField]
+        [Tooltip("Check to enable frame-rate independent smoothing. ")]
+        private bool smoothingActive = true;
```

Comment:
Since we are refactoring, could we also make smoothing params consistent with Solver? On Solvers like Overlap and RadialView, there'sa  "smoothing" checkbox, and then a "move lerp time" and "rotate lerp time" field. See Solver.cs:

```
public static Vector3 SmoothTo(Vector3 source, Vector3 goal, float deltaTime, float lerpTime)
        {
            return Vector3.Lerp(source, goal, lerpTime.Equals(0.0f) ? 1f : deltaTime / lerpTime);
        }
```